### ISSUE TITLE
Hyperparameter Evolution

### ISSUE THREAD

**Issue by zxin8218** [ultralytics/yolov5#3741]
## 🐛 Bug
   ```
ValueError: Total of weights must be greater than zero
```

command
```
python train.py --epochs 10 --weights yolov5x.pt --hyp data/hyp.scratch.yaml --cache --evolve
```
Output:
```
Traceback (most recent call last):
  File "/media/***/DiskA/***/TZ3/yolov5-master/train.py", line 658, in <module>
    main(opt)
  File "/media/***/DiskA/***/TZ3/yolov5-master/train.py", line 614, in main
    x = x[random.choices(range(n), weights=w)[0]]  # weighted selection
  File "/home/***/anaconda3/envs/zx-torch17/lib/python3.9/random.py", line 503, in choices
    raise ValueError('Total of weights must be greater than zero')
```

## Environment
python 3.9
torch 1.7.1

**Comment by glenn-jocher** [comment]
@zxin8218 I've run your command for several generations but am not able to reproduce your error:
<img width="1640" alt="Screenshot 2021-06-23 at 12 46 50" src="https://user-images.githubusercontent.com/26833433/123084285-3dc10000-d421-11eb-8e03-3c7b74e5c9f6.png">


That said there's clearly a random element to the weights values that are passed to your line so it looks like we should first pass the weights through a check to make sure they meet the constraints details in your error message.

**Comment by glenn-jocher** [comment]
@zxin8218 good news 😃! Your original issue may now be fixed ✅ in PR #3748. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
